### ISSUE TITLE
bug: block-opening-brace-space-before-check doesn't behave correctly depending on order of code.

### ISSUE THREAD

**Issue by colinkiama** [vala-lang/vala-lint#156]
There are actually 2 mistakes here. `test2()` and `test3()` however the test code for this is expecting one mistake. False positive?
```vala
void test () { }

void test2 (){ }

void test3 ()
{ }

int main (string[] args) {
    return 0;
}
```
However, if you swap the order of `test2()` and `test3()`, the check works correctly detecting both mistakes. Despite the check behaving correctly, the test code fails.

```vala
void test () { }

void test3 ()
{ }

void test2 (){ }

int main (string[] args) {
    return 0;
}
```

**Comment by colinkiama** [comment]
Consequently, this also affects the behavior of auto-fixes for this check too.
